portal-impl/src/com/liferay/portlet/documentlibrary/util/PDFProcessorImpl.java

```diff
@@ -654,177 +654,207 @@ private void _generateImagesPB(FileVersion fileVersion, File file)
+						processChannel.getProcessNoticeableFuture();
+
+					String processIdentity = String.valueOf(
+						fileVersion.getFileVersionId());
+
+					long pdfBoxTimeout =
+						PropsValues.
+							DL_FILE_ENTRY_PREVIEW_GENERATION_TIMEOUT_PDFBOX;
+
+					if (_log.isDebugEnabled()) {
+						if (generateThumbnail && generatePreview) {
+							_log.debug(
+								StringBundler.concat(
+									"Waiting for ", pdfBoxTimeout,
+									" seconds to generate thumbnail and ",
+									"preview for ", decryptedFile.getPath()));
+						}
+						else {
+							if (generateThumbnail) {
+								_log.debug(
+									StringBundler.concat(
+										"Waiting for ", pdfBoxTimeout,
+										" seconds to generate thumbnail for ",
+										decryptedFile.getPath()));
+							}
+
+							if (generatePreview) {
+								_log.debug(
+									StringBundler.concat(
+										"Waiting for ", pdfBoxTimeout,
+										" seconds to generate preview for ",
+										decryptedFile.getPath()));
+							}
+						}
 					}
 
-					if (generatePreview) {
-						errorMessage =
-							"Timeout when generating preview for " +
-								decryptedFile.getPath();
+					try {
+						future.get(pdfBoxTimeout, TimeUnit.SECONDS);
+
+						futures.put(processIdentity, future);
+					}
+					catch (TimeoutException timeoutException) {
+						String message = null;
+
+						if (generateThumbnail && generatePreview) {
+							message = StringBundler.concat(
+								"Timeout when generating thumbnail and ",
+								"preview for ", decryptedFile.getPath());
+						}
+						else {
+							if (generateThumbnail) {
+								message =
+									"Timeout when generating thumbnail for " +
+										decryptedFile.getPath();
+							}
+
+							if (generatePreview) {
+								message =
+									"Timeout when generating preview for " +
+										decryptedFile.getPath();
+							}
+						}
+
+						if (future.cancel(true)) {
+							message +=
+								" resulted in a canceled timeout for " + future;
+						}
+
+						_fileVersionPreviewEventListener.onFailure(fileVersion);
+
+						if (_log.isWarnEnabled()) {
+							_log.warn(message);
+						}
+
+						throw timeoutException;
 					}
 				}
+				else {
+					LiferayPDFBoxConverter liferayConverter =
+						new LiferayPDFBoxConverter(
+							decryptedFile, thumbnailFile, previewFiles,
+							getPreviewType(fileVersion),
+							getThumbnailType(fileVersion),
+							PropsValues.DL_FILE_ENTRY_PREVIEW_DOCUMENT_DPI,
+							PropsValues.
+								DL_FILE_ENTRY_PREVIEW_DOCUMENT_MAX_HEIGHT,
+							PropsValues.
+								DL_FILE_ENTRY_PREVIEW_DOCUMENT_MAX_WIDTH,
+							generatePreview, generateThumbnail);
 
-				if (future.cancel(true)) {
-					errorMessage +=
-						" resulted in a canceled timeout for " + future;
+					liferayConverter.generateImagesPB();
 				}
-
-				_log.error(errorMessage);
-
-				_fileVersionPreviewEventListener.onFailure(fileVersion);
-
+			}
+			catch (TimeoutException timeoutException) {
```

Comment:
@robertoDiaz Previously, this exception was handled by the general Exception catch. Why do we need to propagate without logging anything now?